app/assets/javascripts/discourse/app/helpers/category-link.js

```diff
@@ -47,7 +48,13 @@ export function categoryBadgeHTML(category, opts) {
   }
 
   const depth = (opts.depth || 1) + 1;
-  if (opts.recursive && depth <= siteSettings.max_category_nesting) {
+  if (opts.ancestors) {
+    const { ancestors, ...otherOpts } = opts;
+    return [category, ...ancestors]
+      .reverse()
+      .map((c) => categoryBadgeHTML(c, otherOpts))
+      .join("");
+  } else if (opts.recursive && depth <= siteSettings.max_category_nesting) {
```

Comment:
Do we still need the `recursive` option? After your change, it is no longer being passed anywhere in core... and I don't see any occurrences in plugins or themes either. (Worth double-checking, though.)

Reply:
On second thought, do we need to pass the ancestors a second time? We have them in the category object, we could still just pass a flag (like `ancestors=true`) and then go through `category.ancestors` and return the html badges for them.

Reply:
We only have them on the category because they are being fetched via `findById`.

Reply:
I'd like to remove the recursive option. I'll double check.